File: Week1/장수희/스택.js
const fs = require("fs");
const input = fs.readFileSync(0, "utf-8").trim().split("\n");

const N = Number(input[0]);

let stack = [];
let result = [];

for (let i = 1; i <= N; i++) {
    const [command, value] = input[i].split(" ");

    if (command === "push") {
      stack.push(Number(value));
    } else if (command === "pop") {
      result.push(stack.length ? stack.pop() : -1);
    } else if (command === "size") {
      result.push(stack.length);
    } else if (command === "empty") {
      result.push(stack.length ? 0 : 1);
    } else if (command === "top") {
      result.push(stack.length ? stack[stack.length - 1] : -1);
    }
  }

console.log(result.join("\n"));

File: Week1/하동헌/스택.py
import sys

N = int(sys.stdin.readline())

stack = []
result = []
for _ in range(N):
    cmd = sys.stdin.readline().split()

    if cmd[0] == "push":
        stack.append(cmd[1])

    elif cmd[0] == "pop":
        if stack:
            result.append(stack.pop())
        else:
            result.append("-1")

    elif cmd[0] == "size":
        result.append(str(len(stack)))

    elif cmd[0] == "empty":
        if stack:
            result.append("0")
        else:
            result.append("1")

    elif cmd[0] == "top":
        if stack:
            result.append(stack[-1])
        else:
            result.append("-1")

print("\n".join(result))
